src/main/scala/apiimport/provider/ApiProviderLike.scala

```diff
@@ -0,0 +1,84 @@
+package apiimport.provider
+
+import java.nio.charset.StandardCharsets.UTF_8
+import java.util.zip.ZipInputStream
+
+import akka.NotUsed
+import akka.stream.scaladsl.Source
+import apiimport.manifests.VoyageManifestParser
+import apiimport.manifests.VoyageManifestParser.VoyageManifest
+import com.typesafe.scalalogging.Logger
+
+import scala.collection.mutable.ArrayBuffer
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Try
+import scala.util.matching.Regex
+
+trait ApiProviderLike {
+  val log: Logger
+  val dqRegex: Regex = "(drt_dq_[0-9]{6}_[0-9]{6})(_[0-9]{4}\\.zip)".r
+
+  def filesAsSource: Source[String, NotUsed]
+
+  def inputStream(file: String): ZipInputStream
+
+  def filterNewer(latestFile: String)(fileName: String): Boolean = fileName >= filterFromFileName(latestFile) && fileName != latestFile
+
+  def filterFromFileName(latestFile: String): String = latestFile match {
+    case dqRegex(dateTime, _) => dateTime
+    case _ => latestFile
+  }
+
+  def manifestsStream(latestFile: String)(implicit ec: ExecutionContext): Source[(String, Try[List[(String, Try[VoyageManifest])]]), NotUsed] = {
+    log.info(s"Requesting DQ zip files > ${latestFile.take(20)}")
```

Comment:
we should probably put some thought into our logging strategy for all new things so we can avoid the same issues we have in our main codebase.